### ISSUE TITLE
Recursive setup expression creates ghost setups that make `VerifyAll` fail

### ISSUE THREAD

**Issue by stakx** [moq/moq4#556]
This has apparently been a problem with Moq for a long time:

```csharp
public interface IFoo
{
    IBar Bar { get; }
}

public interface IBar
{
    int Value { get; set; }
}

var fooMock = new Mock<IFoo>();
fooMock.Setup(foo => foo.Bar.Value).Returns(42).Verifiable("my setup");
var bar = fooMock.Object.Bar;
fooMock.VerifyAll();
```

```
Moq.MockException: The following setups were not matched:
IBar m => m.Value
IBar m => m.Value
my setup: IFoo foo => foo.Bar.Value
```

The first two setups `IBar m => m.Value` should not be there, as there's no corresponding setup for them. (Where do those setups come from?) Their presence makes `VerifyAll` fail.

**Comment by stakx** [comment]
> Where do those setups come from?

The two setups `IBar m => m.Value` are property `get`/`set` setups caused by `SetupAllProperties` caused by traversing the recursive ("fluent") mock expression in `QueryableMockExtensions.FluentMock<T, TResult>(this Mock<T>, Expression<Func<T, TResult>>)`.

This issue would therefore likely be solved by #426.

**Comment by stakx** [comment]
Closing this, this problem will be tracked in #643 (together with a couple other related issues).

**Comment by stakx** [comment]
Due to the decision made in #681 and the corresponding change made to SetupAllProperties in #684, this issue will be fixed in Moq version >4.9.0.
